frontend-react/src/components/FileHandlers/FileHandler.tsx

```diff
@@ -178,12 +179,40 @@ const FileHandler = ({
 
         // initializes necessary state and sets `isSubmitting`
         dispatch({ type: FileHandlerActionType.PREPARE_FOR_REQUEST });
-        await sendFile({
-            contentType: contentType,
-            fileContent: fileContent,
-            fileName: fileName,
-            client: client,
-        });
+
+        let eventData;
+
+        try {
+            const response = await sendFile({
+                contentType: contentType,
+                fileContent: fileContent,
+                fileName: fileName,
+                client: client,
+            });
+
+            eventData = {
+                warningCount: response?.warnings?.length,
+                errorCount: response?.errors?.length,
+            };
+        } catch (e: any) {
+            if (e.data) {
+                eventData = {
+                    warningCount: e.data.warningCount,
+                    errorCount: e.data.errorCount,
+                };
+            }
+        }
+
+        if (eventData) {
```

Comment:
Ah, sorry -- I think we might only want to track this event when `validateOnly` is true too.  Otherwise, we'll be firing it when the user's actually uploading to send rather than just validating.  We'd probably want to track that too, but that's a question for @chris-kuryak.  Will follow up on the ticket!